app/javascript/src/dashboard.js

```diff
@@ -90,9 +89,42 @@ $('document').ready(() => {
   const casaCasePath = id => `/casa_cases/${id}`
   const volunteersTable = $('table#volunteers').DataTable({
     autoWidth: false,
-    stateSave: false,
+    stateSave: true,
+    initComplete: function(settings, json) {
+      this.api().columns().every(function(index) {
+        var columnVisible = this.visible();
+        $('#visibleColumns input[data-column="' + index + '"]').prop('checked', columnVisible);
+      });
+    },
+    stateSaveCallback: function (settings, data) {
+      $.ajax({
+        url: "/table_state_update",
```

Comment:
make sure the urls are `preference_sets/table_state_update` (`rails route -g preference` to see what was created)

Reply:
As I commented above I can't remember the exact naming you gave me. I think this is close, and it passes all test. but let me know if I should change it